acquisitions/src/main/resources/thunderjet/consortia/tenant-utils/tenant.feature

```diff
@@ -0,0 +1,654 @@
+
+    # get tenant details for 'centralTenant' and verify 'setupStatus' will become 'COMPLETED'
+    Given path 'consortia', consortiumId, 'tenants', centralTenant
+    And retry until response.setupStatus == 'COMPLETED'
+    When method GET
+    Then status 200
+    And match response.id == centralTenant
+    And match response.code == 'ABC'
+    And match response.name == 'Central tenants name'
+    And match response.isCentral == true
+    And match response.isDeleted == false
+
+    # verify there is a record for central tenant in 'central_mod_consortia.consortia_configuration'
+    Given path 'consortia-configuration'
+    When method GET
+    Then status 200
+    And match response.centralTenantId == centralTenant
+
+  @Negative
+  # At this point we have one record in consortium = { id: '#(centralTenant)', code: 'ABC', name: 'Central tenants name', isCentral: true }
+  Scenario: Attempt to POST a second tenant with existing 'isCentral' = true, 'id', 'name', 'code'
+    # cases for 409
+    # attempt to create a second central tenant ('isCentral' = true)
+    Given path 'consortia', consortiumId, 'tenants'
+    And request { id: '1234', code: 'XYZ', name: 'Test tenants name', isCentral: true }
+    When method POST
+    Then status 409
+    And match response == { errors : [{ message : 'Object with isCentral [true] is already presented in the system', type : '-1', code: 'DUPLICATE_ERROR' }] }
+
+    # attempt to create a tenant with an existing id
+    Given path 'consortia', consortiumId, 'tenants'
+    And param adminUserId = consortiaAdmin.id
+    And request { id: '#(centralTenant)', code: 'ABE', name: 'test1', isCentral: false }
+    When method POST
+    Then status 409
+    And match response.errors[0].message == 'Object with id [' + centralTenant +'] is already presented in the system'
+    And match response.errors[0].type == '-1'
+    And match response.errors[0].code == 'DUPLICATE_ERROR'
+
+    # attempt to create a tenant with an existing code
+    Given path 'consortia', consortiumId, 'tenants'
+    And param adminUserId = consortiaAdmin.id
+    And request { id: 'non-existing-tenant', code: 'ABC', name: 'test1', isCentral: false }
+    When method POST
+    Then status 409
+    And match response == { errors : [{ message : 'Object with code [ABC] is already presented in the system', type : '-1', code: 'DUPLICATE_ERROR' }] }
+
+    # attempt to create a tenant with an existing name
+    Given path 'consortia', consortiumId, 'tenants'
+    And param adminUserId = consortiaAdmin.id
+    And request { id: 'non-existing-tenant', code: 'ABE', name: 'Central tenants name', isCentral: false }
+    When method POST
+    Then status 409
+    And match response == { errors : [{ message : 'Object with name [Central tenants name] is already presented in the system', type : '-1', code: 'DUPLICATE_ERROR' }] }
+
+  @Negative
```

Comment:
negative tests cases not needed, they are tested in mod-consortia submodule.

Reply:
done